Tasks/Maven/maven.ps1

```diff
@@ -28,43 +40,27 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
-{
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
-    {
-        throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
-    }
-
-    Write-Host "Setting JAVA_HOME to $jdkPath"
-    $env:JAVA_HOME = $jdkPath
-    Write-Verbose "JAVA_HOME set to $env:JAVA_HOME"
-}
+. ./mavenHelper.ps1
 
-Write-Verbose "Running Maven..."
-Invoke-Maven -MavenPomFile $mavenPOMFile -Options $options -Goals $goals
+# Use a specific JDK
+ConfigureJDK $jdkVersion $jdkArchitecture
 
-# Publish test results files
-$publishJUnitResultsFromAntBuild = Convert-String $publishJUnitResults Boolean
-if($publishJUnitResultsFromAntBuild)
-{
-   # check for JUnit test result files
-    $matchingTestResultsFiles = Find-Files -SearchPattern $testResultsFiles
-    if (!$matchingTestResultsFiles)
-    {
-        Write-Host "No JUnit test results files were found matching pattern '$testResultsFiles', so publishing JUnit test results is being skipped."
-    }
-    else
-    {
-        Write-Verbose "Calling Publish-TestResults"
-        Publish-TestResults -TestRunner "JUnit" -TestResultsFiles $matchingTestResultsFiles -Context $distributedTaskContext
-    }    
-}
-else
+# Invoke MVN
+Write-Host "Running Maven..."
+Invoke-Maven -MavenPomFile $mavenPOMFile -Options $options -Goals $goals -ErrorVariable mavenError
+
+if ($mavenError)
```

Comment:
I found a bug here - if the Maven cmdlet fails (for example if the user doesn't set any goal), then the script would continue to run - it would try to publish results etc. So I am making sure the scripts stops if Maven fails. Please let me know if you'd like me to open a bug for this and leave you to track / fix separately.